src/io/save_gals_binary.c

```diff
@@ -96,7 +96,7 @@ int32_t save_binary_galaxies(const int32_t filenr, const int32_t treenr, const i
 
         // Here we move the offset pointer depending upon the number of galaxies processed up to this point.
         struct GALAXY_OUTPUT *galaxy_output = all_outputgals + cumul_output_ngal[snap_idx] + num_gals_processed[snap_idx];
-        status = prepare_galaxy_for_output(filenr, treenr, &halogal[gal_idx], galaxy_output, halos, haloaux, halogal, run_params);
+        status = prepare_galaxy_for_output(treenr, &halogal[gal_idx], galaxy_output, halos, run_params);
```

Comment:
Is there any specific reason why `prepare_galaxy` is called on a galaxy at a time rather than all galaxies?

Reply:
Not specifically.  It's a decision of whether we want to loop over the the function call, or call the function once and loop inside.

Reply:
I think it makes more sense to loop inside the function